src/main/scala/tech/sourced/gemini/Hash.scala

```diff
@@ -229,14 +229,22 @@ class Hash(session: SparkSession,
 
   protected def saveDocFreqToDB(docFreq: OrderedDocFreq, keyspace: String, tables: Tables): Unit = {
     log.warn(s"save document frequencies to DB")
-    CassandraConnector(session.sparkContext).withSessionDo { cassandra =>
-      val cols = tables.docFreqCols
-      val javaMap = docFreq.df.asJava
 
+    CassandraConnector(session.sparkContext).withSessionDo { cassandra =>
+      val docsCols = tables.featuresDocsCols
       cassandra.execute(
```

Comment:
Not sure if it's a good idea to do `execute` on each row.
Maybe it's better to `sc.parallelize(seq).saveToCassandra`.

Reply:
Regarding this may better wait for someone with more insights than me on Spark. 😛

Reply:
Mostly because Cassandra [does not recommend it](https://docs.datastax.com/en/cql/3.3/cql/cql_using/useBatch.html)

> But batches are often mistakenly used in an attempt to optimize performance. Depending on the batch operation, the performance may actually worsen. Some batch operations place a greater burden on the coordinator node and lessen the efficiency of the data insertion. The number of partitions involved in a batch operation, and thus the potential for multi-node accessing, can increase the latency dramatically. In all batching, the coordinator node manages all write operations, so that the coordinator node can pose a bottleneck to completion.

Reply:
I update code using prepared statement to make it a bit better.